docs/src/manual/fe_intro.md

```diff
@@ -1,111 +1,157 @@
-vector with the nodal values and $\underline{f}$ is the force vector. The elements of
-$\underline{K}$ and $\underline{f}$ are given by
+where we call $\underline{K}$ the (tangent) *stiffness matrix*, $\underline{a}$ the *solution
+vector* with the nodal values and $\underline{f}$ the *force vector*. The specific naming is for
+historical reasons, because the finite element method has its origins in mechanics. The elements
+of $\underline{K}$ and $\underline{f}$ are given by
 
 ```math
 \underline{K}_{ij} =
-    \int_\Omega \mathbf{\nabla}N_i \cdot (k \cdot \mathbf{\nabla}N_j) \mathrm{d}\Omega\\
+    \int_{\Omega_\mathrm{h}} \nabla N_i \cdot (k \nabla N_j) \mathrm{d}\Omega \, , \\
 
 \underline{f}_{i} =
-    \int_\Gamma N_i q^\mathrm{p} \mathrm{d}\Gamma + \int_\Omega N_i b \mathrm{d}\Omega
+    \int_{\Gamma_\mathrm{N}} N_i \, q^\mathrm{p} \, \mathrm{d}\Gamma + \int_{\Omega_\mathrm{h}} N_i \, b \, \mathrm{d}\Omega \, .
 ```
 
-The solution to the system (which in this case is linear) is simply given by inverting the
-matrix $\underline{K}$. We also need to take care of the Dirichlet boundary conditions, by
-enforcing the correct nodal values $a_i$ to the prescribed values.
+Finally we also need to take care of the Dirichlet boundary conditions. These are enforce by
+setting the corresponding $a_i$ to the prescribed values and eliminating the associated equations
+from the system. Now, solving this equation system yields an the nodal values and thus an
+approximation to the true solution.
 
-```math
-\underline{a} = \underline{K}^{-1}\ \underline{f}
-```
+## Notes on the Implementation
+
+In practice, the shape functions $N_i$ are only non-zero on parts of the domain $\Omega_\mathrm{h}$.
+Thus, the integrals are evaluated on sub-domains, called *elements* or *cells*.
 
-## Implementation
+TODO: Image of a linear basis function on the grid above.
 
-In practice, the shape functions $N$ are only non-zero on parts of the domain $\Omega$.
-Thus, the integrals are evaluated on sub-domains, called *elements* or *cells*. Each cell
-gives a contribution to the global stiffness matrix and force vector. For a solution of the
-heat equation, as implemented in `Ferrite`, check out
-[this thoroughly commented example](@ref Heat-Equation).
+Each cell gives a contribution to the global stiffness matrix and force vector. The process
+of constructing the system of equations is also called *assembly*. For clarification,
+let us rewrite the formula for the stiffness matrix entries as follows:
+```math
+\underline{K}_{ij}
+    = \int_{\Omega_\mathrm{h}} \nabla N_i \cdot (k \nabla N_j) \mathrm{d}\Omega
+    = \sum_{E \in \Omega_\mathrm{h}} \int_E \nabla N_i \cdot (k \nabla N_j) \mathrm{d}\Omega \, .
+```
+This formulation underlines the element-centric perspective of finite element methods and
+reflects how it is usually implemented in software. For an example of the implementation to
```

Comment:
```suggestion
reflects how it is usually implemented in software. Numerical integration is used 
to integrate over the element. The stiffness matrix entries are then calculated as
```math
\underline{K}_{ij}   
    = \sum_{E \in \Omega_\mathrm{h}} \sum_{\mathrm{ip} \in E} \nabla N_i(x_\mathrm{ip}) \cdot (k \nabla N_j(x_\mathrm{ip})) w_\mathrm{ip} \, .
```where $x_\mathrm{ip}$ are the predefined integration points and $w_\mathrm{ip}$ the corresponding weights.

For an example of the implementation to
```

Reply:
Using it for my teaching already next week was partly a joke (but also serious, especially later)! So don't feel pressured to finish/merge quickly - it was also meant to highlight that this PR will be very useful when we start using ferrite in our teaching. 

For numerical integration, I was thinking about something along the above lines

Reply:
:)

I added a short description about numerical quadrature. If you have suggestions for adjustments and/or a recommendable reference, then I am happy to incorporate it. I can not recommend a reference by-myself, because I learned it the hard way via rigorous math.

Reply:
Nice! I like your updates, also to the heat flow also to make it better connected to the intro. 
At Chalmers, I learned it from "Introduction to the Finite Element Method" by Niels Ottosen and Hans Petersson (1992). My (biased) opinion is that this introduction is quite nice in Chapter 20 (also isoparametric elements chapter 19), so we could refer to that book.